lib/upipe-modules/upipe_rtp_prepend.c

```diff
@@ -167,6 +167,33 @@ static void upipe_rtp_prepend_input(struct upipe *upipe, struct uref *uref,
     ubuf_block_unmap(header, 0);
     upipe_rtp_prepend->seqnum++;
 
+    const char *def;
+    ubase_assert(uref_flow_get_def(upipe_rtp_prepend->flow_def, &def));
+    if (!ubase_ncmp(def, "block.rtp.mp2.sound.") ||
+            !ubase_ncmp(def, "block.rtp.mp3.sound.")) {
+        /* alloc mpa header */
+        struct ubuf *mpa_header = ubuf_block_alloc(uref->ubuf->mgr, 4);
```

Comment:
Unfortunately, you are not allowed to do that :(. The reason is that ubuf->mgr could be a ubuf manager from a hardware demux, with buffer backed by video or special memory, so it may not be able to allocate anything, and it may not be efficient.
The correct way is the more cumbersome urequest with a ubuf manager that we do everywhere else.